.github/workflows/pull-request-workflow.yml

```diff
@@ -53,12 +57,59 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        node-version: [ 16.x ]
+        node-version: [16.x, 18.x]
     steps:
       - uses: actions/checkout@v3
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
         with:
           node-version: ${{ matrix.node-version }}
       - name: Install dependencies
         run: yarn install --immutable
       - name: Build
         run: yarn build
+
+  tests:
+    needs:
+      - build
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [16.x, 18.x]
+    services:
+      postgres:
+        image: postgres
+        env:
+          POSTGRES_PASSWORD: example
+          POSTGRES_DB: test_integration
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+        ports:
+          - 5432:5432
+    steps:
+      - uses: actions/checkout@v3
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Install packages
+        run: yarn install --immutable
+      - name: Run tests
+        run: yarn test
+        env:
+          POSTGRES_HOST: localhost
+          POSTGRES_PASSWORD: example
+          POSTGRES_DB: test_integration
+          POSTGRES_PORT: 5432
+      - uses: ArtiomTr/jest-coverage-report-action@v2
+        id: coverage
+        with:
+          skip-step: all
+          coverage-file: ./**/coverage-final.json
+          output: report-markdown
+      - uses: marocchino/sticky-pull-request-comment@v2
```

Comment:
Also, I see it will duplicate comment since we execute this job twice for different nodejs version, can we make it only once?

Reply:
> Great! Could we please add 18.x nodejs version to each job and connect something like this https://github.com/marketplace/actions/jest-coverage-report ?

I investigated a lot, but I wasn't able to find way to generate one report for all packages in monorepository.

Reply:
Only once for which version?

Reply:
let's do it for stable version